Control/Concurrent/MVar/Strict.hs

```diff
@@ -63,22 +66,27 @@ import Control.DeepSeq
 --     fairness properties of abstractions built using 'MVar's.
 --
 putMVar  :: NFData a => MVar a -> a -> IO ()
-#ifndef __HADDOCK__
-putMVar (MVar mvar#) !x = rnf x `seq` IO $ \ s# -> -- strict!
-    case putMVar# mvar# x s# of
-        s2# -> (# s2#, () #)
-#endif
+putMVar !mv !x = rnf x `seq` MV.putMVar mv x
 
 -- | A non-blocking version of 'putMVar'.  The 'tryPutMVar' function
 -- attempts to put the value @a@ into the 'MVar', returning 'True' if
 -- it was successful, or 'False' otherwise.
 --
 tryPutMVar  :: NFData a => MVar a -> a -> IO Bool
-#ifndef __HADDOCK__
-tryPutMVar (MVar mvar#) !x = IO $ \ s# -> -- strict!
-    case tryPutMVar# mvar# x s# of
-        (# s, 0# #) -> (# s, False #)
-        (# s, _  #) -> (# s, True #)
+tryPutMVar !mv !x = rnf x `seq` MV.tryPutMVar mv x
```

Comment:
Gratuitous bang?
```suggestion
tryPutMVar !mv x = rnf x `seq` MV.tryPutMVar mv x
```

Reply:
Again, needed if we want to force to WHNF if `rnf` does nothing.